### PR TITLE
update http proxy readme

### PR DESCRIPTION
**Pull request by yupbank**
address https://github.com/kubeflow/kubeflow/issues/352

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/363)

### CODE REVIEW

File: components/k8s-model-server/http-proxy/README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ This component provides web server that proxy http rest api to Tensorflow/Servin
 
 The REST Api Spec is similar to Google Cloud Machine Learning Engine Prediction API.
```
Comment (jlewi) [inline components/k8s-model-server/http-proxy/README.md]: Can you add a relevant link to CMLE API?

File: components/k8s-model-server/http-proxy/README.md [hunk 2]
```diff
@@ -43,5 +43,47 @@ or
 
 - **Input and Output Schema**:
 
-In the example above, that correspones to a [tensorflow savedModel](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/saved_model/README.md) with input signature `"tf_model_input"` and `"image"` and output signature `"tf_model_output"`.
+In the example above, that correspones to a [tensorflow savedModel](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/saved_model/README.md) with method signature `tf.saved_model.signature_constants.PREDICT_METHOD_NAME`, input signature `"tf_model_input"` and `"image"` and output signature `"tf_model_output"`.
```
Comment (jlewi) [inline components/k8s-model-server/http-proxy/README.md]: Which example are you referring to?

Comment (jlewi) [inline components/k8s-model-server/http-proxy/README.md]: Can you expand the text? It

Where is the signature def for the PREDICT_METHOD_NAME
Is it here?
https://github.com/tensorflow/serving/blob/master/tensorflow_serving/g3doc/signature_defs.md#classification-signaturedef

What do you mean by input signature "tf_model_input" and "image"

File: components/k8s-model-server/http-proxy/README.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ This component provides web server that proxy http rest api to Tensorflow/Servin
 
 The REST Api Spec is similar to Google Cloud Machine Learning Engine Prediction API.
 
-# API
+# API(rpc proxy)
```
Comment (jlewi) [inline components/k8s-model-server/http-proxy/README.md]: Here's a suggestion for how to maybe layout this document

- Explain why you would want to use HTTP proxy; i.e. that TF serving only supports gRPC so if you want to use HTTP we need a proxy that converts HTTP to gRPC

- Explain the model/TF serving requirements that must be satisfied in order to use HTTP Proxy
  (I think the model must have one of the supported signature methods predict or classify)

- Explain how to send requests;
   What the different methods are for and why to use one or the other.
   Explain the structure of the inputs/outputs
   Explain how to deal with binary data (e.g. b64 data).

File: components/k8s-model-server/http-proxy/README.md (hunk 2 above)
Comment (pdmack) [inline components/k8s-model-server/http-proxy/README.md]: "corresponds"

File: components/k8s-model-server/http-proxy/README.md [hunk 4]
```diff
@@ -43,5 +43,47 @@ or
...
+{"result": 
+	{"classifications": [
+				{"classes": [{"score": 0.0, "label": "0"}, {"score": 1.0, "label": "1"}]},
+				{"classes": [{"score": 0.0, "label": "0"}, {"score": 1.0, "label": "1"}]},
+				]}}
+```
+
+- **Input and Output Schema**:
+
+In the example above, that correspones to a [tensorflow savedModel](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/saved_model/README.md) with method signature `tf.saved_model.signature_constants.CLASSIFY_METHOD_NAME`, input signature `"tf_model_input"` and `"image"` and output signature `"tf_model_output"`.
```
Comment (pdmack) [inline components/k8s-model-server/http-proxy/README.md]: "corresponds"

**Comment by lluunn** [comment]
Any update on this?

**Comment by yupbank** [comment]
ah... sorry i'll update this today later...

File: components/k8s-model-server/http-proxy/README.md (hunk 2 above)
Comment (yupbank) [reply components/k8s-model-server/http-proxy/README.md]: the rest api example. i'll add a link to that

Comment (yupbank) [reply components/k8s-model-server/http-proxy/README.md]: > Where is the signature def for the PREDICT_METHOD_NAME
that's a constant defined in tensorflow `tensorflow.saved_model.signature_constants.PREDICT_METHOD_NAME`


input params with signature/key "tf_model_input" and "image"

**Comment by lluunn** [comment]
please @ me if it's ready for review, thanks!

**Comment by yupbank** [comment]
@lluunn  hey, i think it's ready for some 👀

**Comment by lluunn** [comment]
looking now. you need to merge master to fix the test

**Comment by lluunn** [comment]
Mostly looks good to me.
Could you add some word on the model_name / version_name?
For /model/${model_name}/version/${model_version}:predict

Why do we have model/version, what's the model_name and model_version, how do we determine the name, etc.
Or pointer to TF serving on this topic.

**Comment by yupbank** [comment]
Sure... let me update that

**Pull request merged**